src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -236,21 +238,21 @@ void primitive_inst::realloc_if_needed() {
     if (_node.is_type<input_layout>())
         return;
 
-    bool can_reuse_buffer = _output && actual_layout.count() <= max_output_layout_size;
+    bool can_reuse_buffer = _outputs[0] && actual_layout.count() <= max_output_layout_size;
 
     if (can_reuse_buffer) {
         GPU_DEBUG_IF(debug_config->verbose >= 4) {
             GPU_DEBUG_COUT << id() << ": reuse previously allocated output buffer" << std::endl;
         }
-        _output = _network.get_engine().reinterpret_buffer(*_output, actual_layout);
+        _outputs[0] = _network.get_engine().reinterpret_buffer(*_outputs[0], actual_layout);
```

Comment:
I thinks this should be checked and done for all "outputs"

Reply:
For this, Updating program_node APIs and kernel_impl_params related to output layouts should be implemented first. So would it be okay to update it together through the next PR?